dataprofiler/tests/profilers/test_profile_builder.py

```diff
@@ -1927,6 +1927,97 @@ def test_unique_row_ratio_empty_profiler(self):
         profiler = StructuredProfiler(pd.DataFrame([]))
         self.assertEqual(0, profiler._get_unique_row_ratio())
 
+    def test_null_replication_metrics_calculation(self):
+        data = pd.DataFrame(
+            {
+                "a": [3, 2, np.nan, 7, np.nan],
+                "b": [10, 10, 1, 4, 2],
+                "c": [2, 5, 3, 5, 7],
+            }
+        )
+        profile_options = dp.ProfilerOptions()
+        profile_options.set(
+            {
+                "data_labeler.is_enabled": False,
+                "structured_options.multiprocess.is_enabled": False,
+                "null_replication_metrics.is_enabled": True,
+            }
+        )
+        profiler = dp.StructuredProfiler(data, options=profile_options)
+        report = profiler.report()
+
+        self.assertTrue("null_replication_metrics" in report["data_stats"][0])
+        column = report["data_stats"][0]["null_replication_metrics"]
+
+        self.assertTrue(len(column["class_prior"]) == 2)
+        self.assertTrue(len(column["class_mean"]) == 2)
+
+        np.testing.assert_array_almost_equal([3 / 5, 2 / 5], column["class_prior"])
+
+        np.testing.assert_array_almost_equal([24, 12], column["class_sum"][0])
+        np.testing.assert_array_almost_equal([3, 10], column["class_sum"][1])
+
+        np.testing.assert_array_almost_equal([24 / 3, 12 / 3], column["class_mean"][0])
+        np.testing.assert_array_almost_equal([3 / 2, 10 / 2], column["class_mean"][1])
+
+        # Test Profile updates
+        data_2 = pd.DataFrame(
+            {
+                "a": [3, 2, np.nan, 7, 5],
+                "b": [10, 10, 1, 4, 2],
+                "c": [2, 5, 3, np.nan, np.nan],
+            }
+        )
+
+        profiler.update_profile(data_2)
```

Comment:
shouldn't the values after this and the values after `merge` be the same?

Reply:
Initially `profiler` has `data`, then update `profiler` with `data_2` (so `profiler` has `data` and `data_2`). Then initialize `profiler2` with `data` then merge it with `profiler`, therefore `merged_profile` contains `2*data+data_2` whereas `profile` after update contains `data+data_2` so it shouldn't be the same.

Reply:
That makes sense. I thought it was going to be `data + data2`

I might suggest doing that as we should validate that merge / update should provide the same results.

one way would be to do merge prior to update to validate.